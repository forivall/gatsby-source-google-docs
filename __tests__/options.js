const documentTexts = require("./documents/texts.json")
const {GoogleDocument} = require("../utils/google-document")

test(`"DemoteHeading" option`, () => {
  const googleDocument = new GoogleDocument(
    documentTexts,
    {},
    {demoteHeadings: true}
  )
  const documentObject = googleDocument.toObject()
  expect(documentObject).toMatchSnapshot()
})

test(`Crosslinks between documents`, () => {
  const options = {
    crosslinksPaths: {
      [documentTexts.documentId]: "/relative-path",
    },
  }
  const googleDocument = new GoogleDocument(documentTexts, {}, options)
  const documentObject = googleDocument.toObject()
  expect(documentObject).toMatchSnapshot()
})
