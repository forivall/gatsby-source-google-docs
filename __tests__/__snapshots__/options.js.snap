// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"DemoteHeading" option 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "h2",
      "value": "Title level 1",
    },
    Object {
      "type": "h3",
      "value": "Title level 2",
    },
    Object {
      "type": "h4",
      "value": "Title level 3",
    },
    Object {
      "type": "h5",
      "value": "Title level 4",
    },
    Object {
      "type": "h6",
      "value": "Title level 5",
    },
    Object {
      "type": "h6",
      "value": "Title level 6",
    },
    Object {
      "type": "p",
      "value": "**bold**",
    },
    Object {
      "type": "p",
      "value": "_italic_",
    },
    Object {
      "type": "p",
      "value": "<ins>underline</ins>",
    },
    Object {
      "type": "p",
      "value": "~~strikethrough~~ ",
    },
    Object {
      "type": "p",
      "value": "<sup>superscript</sup>",
    },
    Object {
      "type": "p",
      "value": "<sub>subscript</sub>",
    },
    Object {
      "type": "p",
      "value": "**_boldItalic_**",
    },
    Object {
      "type": "p",
      "value": "text with **space bold after** and _space italic before_.",
    },
    Object {
      "type": "p",
      "value": "		indented text",
    },
    Object {
      "type": "p",
      "value": "[Link to itself](https://docs.google.com/document/d/1vKqAsTALHsU6AviqRVqofzhfYOr29JLo4Zoemz9fRUw)",
    },
  ],
  "metadata": Object {},
}
`;

exports[`Crosslinks between documents 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "h1",
      "value": "Title level 1",
    },
    Object {
      "type": "h2",
      "value": "Title level 2",
    },
    Object {
      "type": "h3",
      "value": "Title level 3",
    },
    Object {
      "type": "h4",
      "value": "Title level 4",
    },
    Object {
      "type": "h5",
      "value": "Title level 5",
    },
    Object {
      "type": "h6",
      "value": "Title level 6",
    },
    Object {
      "type": "p",
      "value": "**bold**",
    },
    Object {
      "type": "p",
      "value": "_italic_",
    },
    Object {
      "type": "p",
      "value": "<ins>underline</ins>",
    },
    Object {
      "type": "p",
      "value": "~~strikethrough~~ ",
    },
    Object {
      "type": "p",
      "value": "<sup>superscript</sup>",
    },
    Object {
      "type": "p",
      "value": "<sub>subscript</sub>",
    },
    Object {
      "type": "p",
      "value": "**_boldItalic_**",
    },
    Object {
      "type": "p",
      "value": "text with **space bold after** and _space italic before_.",
    },
    Object {
      "type": "p",
      "value": "		indented text",
    },
    Object {
      "type": "p",
      "value": "[Link to itself](/relative-path)",
    },
  ],
  "metadata": Object {},
}
`;
